pkg/handler/iam_handler.go

```diff
@@ -181,37 +202,33 @@ func (h *IAMHandler) handlePolicy(ctx context.Context, p *v1alpha1.ResourcePolic
 	return &v1alpha1.IAMResponse{Resource: p.Resource, Policy: np}, nil
 }
 
-// Remove expired bindings and add or update new bindings with expiration condition.
-func (h *IAMHandler) updatePolicy(ctx context.Context, p *iampb.Policy, bs []*v1alpha1.Binding, expiry time.Time) {
-	logger := logging.FromContext(ctx)
+// addAndCleanupPolicy adds new bindings with expiration condition, it also does
+// best effort cleanup which removes any expired AOD bindings, however any
+// errors encounterred during removal will be ignored and policy update for the
+// request will continue. Removal errors should be handled separately such as
+// in a global IAM cleanup.
+func (h *IAMHandler) addAndCleanupPolicy(ctx context.Context, p *iampb.Policy, bs []*v1alpha1.Binding, expiry time.Time) {
+	// Cleanup policy.
+	h.cleanupExpiredBindings(ctx, p, bs, expiry)
```

Comment:
This func should return an error and log warn here instead.

Reply:
Updated.